### PR TITLE
Provide access to known shard types #23912

### PR DESCRIPTION
**Pull request by danischroeter**

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Resolves #23912

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala [hunk 1]
```diff
@@ -446,6 +452,16 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
   }
 
   /**
+    * Scala API: get all defined sharding typeName's.
```
Comment (ktoso) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: Small rewording please: "Get all currently defined sharding type names"

**Review by ktoso — APPROVED** [review]
LGTM, good idea -- small rewording requested though (`'s` is possessive)

**Comment by ktoso** [comment]
Would it be good to add some simple test for this?

**Comment by raboof** [comment]
PLS BUILD

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala [hunk 2]
```diff
@@ -446,6 +452,16 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
...
+    */
+  def shardTypeNames: immutable.Set[String] = regions.keySet().asScala.toSet
+
+  /**
+    * Java API: get all defined sharding typeName's.
```
Comment (raboof) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: Unfortunately our formatter prefers these lines to be formatted differently. Compiling from sbt should automatically fix the formatting, could you try that?

**Review by raboof — DISMISSED** [review]
I also saw a whitesource failure in jenkins but I don't see that locally, so probably nothing?

**Pull request merged**